ts/features/idpay/wallet/saga/handleInitiativePairing.ts

```diff
@@ -0,0 +1,94 @@
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { call, put } from "typed-redux-saga/macro";
+import { PreferredLanguageEnum } from "../../../../../definitions/backend/PreferredLanguage";
+import { SagaCallReturnType } from "../../../../types/utils";
+import { getGenericError, getNetworkError } from "../../../../utils/errors";
+import { readablePrivacyReport } from "../../../../utils/reporters";
+import { IDPayWalletClient } from "../api/client";
+import {
+  IdpayInitiativesPairingPayloadType,
+  IdpayInitiativesUnpairPayloadType,
+  idpayInitiativesPairingDelete,
+  idpayInitiativesPairingPut
+} from "../store/actions";
+import { showToast } from "../../../../utils/showToast";
+
+type ClientWithProps =
+  | {
+      client: IDPayWalletClient["deleteInstrument"];
+      props: IdpayInitiativesUnpairPayloadType;
+      action: typeof idpayInitiativesPairingDelete;
+    }
+  | {
+      client: IDPayWalletClient["enrollInstrument"];
+      props: IdpayInitiativesPairingPayloadType;
+      action: typeof idpayInitiativesPairingPut;
+    };
+
+export function* handleInitiativePairing(
```

Comment:
Please avoid "swiss knife" functions. I would consider splitting this function in two. Common code could be handled with a callback function.